libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -590,40 +570,30 @@ protected void onPostExecute(OAuth2.TokenEndpointResponse tr) {
                 mgr.getAdminPermsManager().setPrefs(id.customPermissions, account);
             }
 
+            // Save the user account
+            addAccount(account);
+
             // Screen lock required by mobile policy.
             if (id.screenLockTimeout > 0) {
 
                 // Stores the mobile policy for the org.
                 final PasscodeManager passcodeManager = mgr.getPasscodeManager();
                 passcodeManager.storeMobilePolicyForOrg(account, id.screenLockTimeout * 1000 * 60, id.pinLength);
                 passcodeManager.setTimeoutMs(id.screenLockTimeout * 1000 * 60);
-                boolean changeRequired = passcodeManager.setMinPasscodeLength((Activity) getContext(), id.pinLength);
-
-                /*
-                 * Checks if a passcode already exists. If a passcode has NOT
-                 * been created yet, the user is taken through the passcode
-                 * creation flow, at the end of which account data is encrypted.
-                 */
-                if (!passcodeManager.hasStoredPasscode(mgr.getAppContext())) {
-
-                    // This will bring up the create passcode screen - we will create the account in onResume.
-                    passcodeManager.setEnabled(true);
-                    passcodeManager.lockIfNeeded((Activity) getContext(), true);
-                } else if (!changeRequired) {
-
-                    // If a passcode change is required, the lock screen will have already been set in setMinPasscodeLength.
-                    final UserAccount addedAccount = addAccount();
-                    callback.finish(addedAccount);
-                }
+                // NB setMinPasscodeLength(...)
+                //    If there was a passcode and the length is increased, the passcode manager will remember that a passcode change is required
+                //    The next SalesforceActivity to resume, will cause the locking screen to popup in passcode change mode
```

Comment:
We are not triggering the passcode screen here. Instead we leave the passcode manager in a state that means, that next time a SalesforceActivity resumes, the lock screen will be shown (with the change passcode flow).